incubator/monochart/templates/_helpers.tpl

```diff
@@ -64,8 +64,9 @@ env:
 {{/*
 https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#use-pod-fields-as-values-for-environment-variables
 */}}
-{{- with $root.Values.envFromFieldRefFieldPath }}
+{{- if or (not (empty $root.Values.envFromFieldRefFieldPath)) (not (empty $root.Values.envFromSecretKeyRef)) }}
```

Comment:
You need to consider https://github.com/cloudposse/charts/blob/37a41f5599368ae7720ca8d3330b1e305d5f873f/incubator/monochart/templates/_helpers.tpl#L57-L63

And you can use a more concise syntax like https://github.com/cloudposse/charts/blob/37a41f5599368ae7720ca8d3330b1e305d5f873f/incubator/monochart/templates/_helpers.tpl#L26

- `empty` is redundant, conditionals calculate emptiness for you

Reply:
I really didn't like my `if` statement there, but missed the better syntax. Updated that.

Is the forced quoting of values for safety? This is all part of the following Kubernetes features:
https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables

This should all be Kubernetes field paths and secret names and safe for YAML without unnecessary quoting. I'm always a bit fuzzy on this, the rules around quoting in YAML seem inconsistent to me everytime I read them.

Reply:
Added in the string safety after testing it. Seems Kubernetes doesn't mind the strings there, in my limited testing.